pac4j-http/src/test/java/org/pac4j/http/client/direct/CookieClientTests.java

```diff
@@ -0,0 +1,96 @@
+/*
+ *    Copyright 2012 - 2015 pac4j organization
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package org.pac4j.http.client.direct;
+
+import org.apache.commons.codec.binary.Base64;
+import org.junit.Test;
+import org.pac4j.core.context.Cookie;
+import org.pac4j.core.context.MockWebContext;
+import org.pac4j.core.exception.RequiresHttpAction;
+import org.pac4j.core.profile.UserProfile;
+import org.pac4j.core.util.TestsConstants;
+import org.pac4j.core.util.TestsHelper;
+import org.pac4j.http.credentials.TokenCredentials;
+import org.pac4j.http.credentials.authenticator.TokenAuthenticator;
+import org.pac4j.http.credentials.authenticator.test.SimpleTestTokenAuthenticator;
+import org.pac4j.http.profile.creator.test.SimpleTestUsernameProfileCreator;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * This class tests the {@link CookieClient} class.
+ *
+ * @author Misagh Moayyed
+ * @since 1.8.0
+ */
+public class CookieClientTests implements TestsConstants {
+
+    @Test
+    public void testClone() {
+        final CookieClient oldClient = new CookieClient();
+        oldClient.setName(TYPE);
+        final SimpleTestUsernameProfileCreator profileCreator = new SimpleTestUsernameProfileCreator();
```

Comment:
This profile creator does not exist anymore as the authenticator saves its profile in the credentials for the default `AuthenticatorProfileCreator`.